docs/adr/ADR-01-STRUCTURED_DATA.md

```diff
@@ -0,0 +1,148 @@
+
+Pros:
+* Users can introduce their own properties with formats
+
+Cons:
+* Even simple strings have to have a format prefix, otherwise we cannot guarantee correct handling
+* Unclear how to make variants of the same value, since the key would be the same
+
+#### Encode the format into the key
+
+Put the format information somewhere in the key, for example as a suffix
+
+Pros:
+* Users can introduce their own properties with formats
+* Easy to have multiple variants, since the key is different
+
+Cons:
+* Makes key-handling more complex, since there are now multiple keys for one "base-key"
+
+
+### Decision
+
+We decided on putting the format and variant information into the key, as different key=value pairs seperated by ":"
+
+At this point we need to introduce some more precise terminology:
+
+TODO move or copy this glossary somewhere else?
+* Entry: One generic entry in our database, which consists of multiple "Key"->"Value" mappings, where key and value are strings.
+* Event: A specialised "Entry", which represents an event and has two mandatory keys: "name" and "startDate"
+* Key: Key value of one "Key"->"Value" mapping of one entry. Can contain multiple variant information. Format is "PropertyName"\{:"VariantInformation"}
+* PropertyName: Name of the property of an entry. Can only consist of letters, numbers and "." TODO is that ok? do we want to allow other special chars?
+* VariantInformation: Specifying what kind of variant this key is. Format is "VariantName"="VariantValue"
+
+Currently supported variants are:
+* Language: which language a value is in
+* Format: what kind of format the value is in
+
+If a certain variant makes sense for a certain property needs to be decided per property basis.
+
+#### Language Variant
+
+Has the VariantName "lang" and the value is a two-letter language abbreviation, like "en", "de", "fr", ...
+
+TODO define what that means for searching, for displaying, how languages are chosen and so on
+
+#### Format Variant
+
+Has the VariantName "format" and for the value see the [Formats](#formats) section.
+
+TODO define more stuff here?
+
+
+## Formats
+
+We decided to support following formats for now:
+* Numbers (VariantValue "number")
+* Dates (VariantValue "date")
+* Lists (VariantValue "list")
+* Markdown (VariantValue "markdown")
+
+### Numbers
+
+Knowledge about numbers is mostly important to correctly sort them.
+
+The format looks like: TODO define format and possible numbers here
+
+Examples:
+* 0
+* 1
+* -5
+* ...?
+
+### Dates
+
+Knowledge about dates is important for before/after + duration searches and to sort them correctly.
+
+The format is the "Date and time with the offset" format of the [ISO 8601](https://en.wikipedia.org/wiki/ISO_8601) standard.
+
+Examples:
+* 2024-04-27T23:59:00+02:00
+* 2024-04-27T11:00:00Z
+
+### Lists
+
+Knowledge about lists is mostly important for contains searches and search faceting. We cannot sort lists. (TODO can/should we?)
```

Comment:
sorting according to list values should be done by client, not?
it would be more complex as we would also need some information about the sort order. maybe we have to implement this later when we find a use case for it in the publisher, but for now i would not